### PR TITLE
Laboratory work #1, Baranov Timofey - 22FPL1

### PR DESCRIPTION
**Pull request by tima-baranov**

### CODE REVIEW

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Hm.. Ok, score was changed. What about code?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
waiting for implementation...

**Comment by artyomtugaryov** [comment]
Change score is not enough:(

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -11,6 +11,14 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+
+    tokens = text.lower()
+    for symbol in text:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:18]: Le't create a list and add to it if symbol is alpha? It should be easier

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (tima-baranov) [reply lab_1_classify_by_unigrams/main.py:19]: Probably fixed..

Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:12]: Are you sure?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (tima-baranov) [reply lab_1_classify_by_unigrams/main.py:12]: No :(

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -2,20 +2,19 @@
 Lab 1
 Language detection
 """
-import json
 
 
 def tokenize(text: str) -> list[str] | None:
-    """
```
[the remark this answers is not in the document]
Comment (tima-baranov) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -2,20 +2,19 @@
...
-    :return: a list of lower-cased tokens without punctuation
-    """
     if not isinstance(text, str):
         return None
+    punctuation = "1234567890!@#№;$%:^&?*()-_=+[,{}]'<>/\\`.~"
+    tokens = text.lower()
+    for token in tokens:
+        if token in punctuation:
+            tokens = tokens.replace(token, '')
+            tokens = tokens.replace(' ', '')
```
[the remark this answers is not in the document]
Comment (tima-baranov) [reply lab_1_classify_by_unigrams/main.py]: Fixed

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -2,20 +2,19 @@
...
 def tokenize(text: str) -> list[str] | None:
-    """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation, digits and other symbols
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
     if not isinstance(text, str):
         return None
+    punctuation = "1234567890!@#№;$%:^&?*()-_=+[,{}]'<>/\\`.~"
```
[the remark this answers is not in the document]
Comment (tima-baranov) [reply lab_1_classify_by_unigrams/main.py]: Fixed

**Pull request closed without merging**